### PR TITLE
sdk/state: add some helpful funcs

### LINKED ISSUE #175: sdk/state: add Equal funcs to OpenAgreement and CloseAgreement
One thing I discovered working on the #122 is that it would be helpful to have `Equal` functions on the `OpenAgreement` and `CloseAgreement` types. I suspect we also will need them for #163, but we also need it for users of the SDK.

### PR DESCRIPTION
**Pull request by leighmcculloch**
### What
Add some functions that I've found helpful:
 - Equal functions on agreements.
 - Functions to get the escrow account objects from the channel.

### Why
It's helpful for the user to be able to compare agreements to know if an agreement they have is the same as a new agreement.

It's helpful to be able to access the full state of the escrow account as known by the channel, and not just the balance that we expose now.

I discovered this while working on #122.

Close #175 

### Known Limitations
I shortcutted writing the equals methods by using the cmp.Equal package. We should really write a proper equals package that doesn't use reflection under the surface, but this should be sufficient for now.

### CODE REVIEW

File: sdk/state/open_test.go [hunk 1]
```diff
@@ -1,14 +1,131 @@
...
 	"time"
 
 	"github.com/stellar/go/keypair"
 	"github.com/stellar/go/network"
+	"github.com/stellar/go/xdr"
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
+func TestOpenAgreement_Equal(t *testing.T) {
```
Comment (acharb) [inline sdk/state/open_test.go:15]: 💭 `cmp` compares signatures as well? In which case can we add a test case of 2 different signatures to

**Review by acharb — APPROVED** [review]
lgtm, left 1 💭

File: sdk/state/open_test.go (hunk 1 above)
Comment (leighmcculloch) [reply sdk/state/open_test.go:15]: Yup. The third last and last test case test that actually 😄.

Comment (acharb) [reply sdk/state/open_test.go:15]: ah I meant two existing sigs that are different, not existent and non-existent

I don't see that case being tested 🤔

Comment (leighmcculloch) [reply sdk/state/open_test.go:15]: Oh gotcha. That's what the last test case was supposed to be but looks like it has a bug. I'll fix it in another PR.

**Pull request merged**